src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32Program.cs

```diff
@@ -928,42 +840,41 @@ private class RemoveDuplicatesComparer : IEqualityComparer<Win32Program>
 
             public bool Equals(Win32Program app1, Win32Program app2)
             {
-                if (!string.IsNullOrEmpty(app1.Name) && !string.IsNullOrEmpty(app2.Name)
-                    && !string.IsNullOrEmpty(app1.ExecutableName) && !string.IsNullOrEmpty(app2.ExecutableName)
-                    && !string.IsNullOrEmpty(app1.FullPath) && !string.IsNullOrEmpty(app2.FullPath))
+                if (app1 == null || app2 == null)
                 {
-                    // Using OrdinalIgnoreCase since this is used internally
-                    return app1.Name.Equals(app2.Name, StringComparison.OrdinalIgnoreCase)
-                        && app1.ExecutableName.Equals(app2.ExecutableName, StringComparison.OrdinalIgnoreCase)
-                        && app1.FullPath.Equals(app2.FullPath, StringComparison.OrdinalIgnoreCase);
+                    return false;
                 }
 
-                return false;
+                return (app1.FullPath.ToUpperInvariant(), app1.HasArguments ? app1.Arguments.ToUpperInvariant() : null)
+                    .Equals((app2.FullPath.ToUpperInvariant(), app2.HasArguments ? app2.Arguments.ToUpperInvariant() : null));
```

Comment:
Should we handle a case when FullPath or Arguments are null?

Reply:
HasArguments does check for null